### PR TITLE
HV-1147 Some adjustments to enable usage of the patching mechanism on…

### PR DESCRIPTION
**Pull request by gunnarmorling**
… JDK 9

### CODE REVIEW

File: integration/pom.xml [hunk 1]
```diff
@@ -214,6 +214,18 @@
                         <artifactId>wildfly-patching</artifactId>
                         <version>1.0.1.Final</version>
                     </dependency>
+                    <!-- The exclusion is needed to have the build work with JDK 9 -->
+                    <dependency>
+                        <groupId>org.wildfly.core</groupId>
+                        <artifactId>wildfly-cli</artifactId>
+                        <version>1.0.1.Final</version>
+                        <exclusions>
+                            <exclusion>
```
Comment (gsmet) [inline integration/pom.xml]: Couldn't you put the exclusion directly on the wildfly-patching artifact?

**Comment by gunnarmorling** [comment]
Pushed another commit which makes the _documentation_ module pass, too. Only remaining issue is in _distribution_ related to JavaDocs.

**Comment by gsmet** [comment]
> * amending Surefire arg line with required module exports

I don't see this part in the commit? Did I miss something?

And did you test locally (on a multi-core machine) that removing the 4 threads for the engine tests doesn't slow down the build?

**Comment by gunnarmorling** [comment]
> I don't see this part in the commit? Did I miss something?

I happens by virtue of the `surefire.argLine` and `surefire.argLine.extension` properties in the parent POM. There is a "jdk-9" profile which does it once for all sub-modules.

> And did you test locally (on a multi-core machine) that removing the 4 threads for the engine tests doesn't slow down the build?

Ah, seems I went a bit too far by trying to centralize the Surefire config. I've added this back.

I've also pushed another commit which makes the distribution build work. JavaDoc can be generated now on 9 without a build failure, but it looks broken. I reckon the style sheet was changed in Java 9 again. I'll leave that for another time (going to open an issue).

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**